File: app/src/main/java/ru/merkulyevsasha/news/presentation/splash/SplashActivity.java
package ru.merkulyevsasha.news.presentation.splash;


import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import ru.merkulyevsasha.news.R;
import ru.merkulyevsasha.news.presentation.main.MainActivity;


public class SplashActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        initDownloadChannel();

        Intent intent = new Intent(this, MainActivity.class);
        startActivity(intent);
        finish();
    }

    private void initDownloadChannel() {
        // Create the NotificationChannel, but only on API 26+ because
        // the NotificationChannel class is new and not in the support library
        // Creating an existing notification channel with its original values performs no operation, so it's safe to call this code when starting an app.
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            CharSequence name = getString(R.string.download_channel_name);
            String description = getString(R.string.download_channel_description);
            int importance = NotificationManager.IMPORTANCE_DEFAULT;
            NotificationChannel channel = new NotificationChannel(getString(R.string.notification_channell_id), name, importance);
            channel.setDescription(description);
            // Register the channel with the system; you can't change the importance
            // or other notification behaviors after this
            NotificationManager notificationManager = getSystemService(NotificationManager.class);
            if (notificationManager != null) {
                notificationManager.createNotificationChannel(channel);
            }
        }
    }
}

File: app/src/main/java/ru/merkulyevsasha/news/presentation/splash/SplashActivity.kt
package ru.merkulyevsasha.news.presentation.splash

import android.app.NotificationChannel
import android.app.NotificationManager
import android.os.Build
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.widget.Toast
import ru.merkulyevsasha.RequireServiceLocator
import ru.merkulyevsasha.core.ServiceLocator
import ru.merkulyevsasha.core.domain.SetupInteractor
import ru.merkulyevsasha.news.R
import java.util.*

class SplashActivity : AppCompatActivity(), SplashView, RequireServiceLocator {

    private var presenter: SplashPresenterImpl? = null
    private lateinit var serviceLocator: ServiceLocator

    override fun setServiceLocator(serviceLocator: ServiceLocator) {
        this.serviceLocator = serviceLocator
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        registerNewsChannel()

        val setupInteractor = serviceLocator.get(SetupInteractor::class.java)
        presenter = SplashPresenterImpl(setupInteractor)
    }

    override fun showMainScreen() {
//        serviceLocator.getApplicationRouter().showMainActivity()
        finish()
    }

    override fun showFatalError() {
        Toast.makeText(this, "Oops!", Toast.LENGTH_LONG).show()
        finish()
    }

    override fun onResume() {
        super.onResume()
        presenter?.bindView(this)
        presenter?.onSetup(getFirebaseId = { UUID.randomUUID().toString() })
    }

    override fun onPause() {
        presenter?.unbindView()
        super.onPause()
    }

    override fun onDestroy() {
        presenter?.onDestroy()
        presenter = null
        serviceLocator.releaseAll()
        super.onDestroy()
    }

    private fun registerNewsChannel() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val name = getString(R.string.download_channel_name)
            val description = getString(R.string.download_channel_description)
            val importance = NotificationManager.IMPORTANCE_DEFAULT
            val channel = NotificationChannel(getString(R.string.notification_channell_id), name, importance)
            channel.description = description
            // Register the channel with the system; you can't change the importance
            // or other notification behaviors after this
            val notificationManager = getSystemService(NotificationManager::class.java)
            notificationManager?.createNotificationChannel(channel)
        }
    }
}
